src/DependencyInjection/DoctrineExtension.php

```diff
@@ -1451,8 +1451,22 @@ private function registerDbalMiddlewares(
         $loader->load('middlewares.php');
 
         $loggingMiddlewareAbstractDef = $container->getDefinition('doctrine.dbal.logging_middleware');
+
         foreach ($connWithLogging as $connName) {
+            // Register logging middlewares only when a logger service is available
+            if (! $container->has('logger')) {
+                continue;
+            }
```

Comment:
Shouldn't it be moved below? Otherwise doctrine.middleware tag is not being added to logging middleware

Reply:
Done in 9c72fc33 and bab865a9